homeassistant/components/transmission/__init__.py

```diff
@@ -181,19 +178,31 @@ def add_torrent(service):
                 )
 
         self.hass.services.async_register(
-            DOMAIN, SERVICE_ADD_TORRENT, add_torrent, schema=SERVICE_ADD_TORRENT_SCHEMA
+            DOMAIN, self.service_name, add_torrent, schema=SERVICE_ADD_TORRENT_SCHEMA
         )
 
         self.config_entry.add_update_listener(self.async_options_updated)
 
         return True
 
+    def add_options(self):
+        """Add options for entry."""
+        if not self.config_entry.options:
+            scan_interval = self.config_entry.data.pop(
+                CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
```

Comment:
We shouldn't make scan interval customizable.

Reply:
Scan interval is an option. If it is imported from config.yaml it is added to options otherwise the default value is used. This option already exists in the current code.

Reply:
It should not be part of the config flow.

Reply:
Its not. You can check the config flow form to confirm.

Reply:
It will only be part of the config_entry.data if imported from config.yaml

Reply:
oh I see, ok .